pipelines/base/config-dump.yaml

```diff
@@ -0,0 +1,139 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: dump-config
+spec:
+  params:
+  - name: REPO_NAME
+    type: string
+    default: .github
+  - name: INSTALLATION_ID
+    type: string
+  - name: GIT_EMAIL
+    type: string
+  - name: GIT_NAME
+    type: string
+  steps:
+    - name: dump-and-push-config
+      image: quay.io/operate-first/opf-toolbox
+      volumeMounts:
+        - mountPath: /var/lib/shared
+          name: shared-data
+      env:
+        - name: GITHUB_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: peribolos-$(params.INSTALLATION_ID)
+              key: token
+        - name: ORG_NAME
+          valueFrom:
+            secretKeyRef:
+              name: peribolos-$(params.INSTALLATION_ID)
+              key: orgName
+      script: |
+        #!/bin/bash
+        REPO_REMOTE="https://$GITHUB_TOKEN@github.com/$ORG_NAME/$(params.REPO_NAME)"
+
+        DEFAULT_BRANCH=$( curl \
+          -H "Accept: application/vnd.github.v3+json" \
+          -H "Authorization: Bearer ${GITHUB_TOKEN}" \
+          https://api.github.com/repos/$ORG_NAME/$(params.REPO_NAME) 2>/dev/null \
+          | yq e '.default_branch' - | sed 's/\"//g' )
+        echo -n $DEFAULT_BRANCH > /var/lib/shared/default_branch
+
+        BRANCH_NAME="$(context.taskRun.name)"
+
+        git config --global user.email "$(params.GIT_EMAIL)"
+        git config --global user.name "$(params.GIT_NAME)"
+
+        mkdir repo
+        cd repo
+
+        # Check if repo has no commits
+        if [ $( git ls-remote --heads $REPO_REMOTE | wc -l ) -eq "0" ]; then
+          echo "Repository empty, creating empty commit"
+          git init -b $DEFAULT_BRANCH
+          git remote add origin $REPO_REMOTE
+          git commit --allow-empty -m "Initial commit"
+          git push --all origin
+        else
+          git clone $REPO_REMOTE .
+        fi
+        git checkout -b $BRANCH_NAME
+
+        echo -n $GITHUB_TOKEN > token
+        echo "Dumping org config ..."
+        peribolos --dump $ORG_NAME --github-token-path token > config.yaml
+
+        echo "Commiting config to $BRANCH_NAME branch ..."
+        git add config.yaml
```

Comment:
I've noticed @mimotej  named this file `github-config.yaml` in his push-event task. We need to be consistent... Either name is fine I think, we just need to keep the file the same name.

Reply:
I didn't notice that thanks! Renamed to `github-config.yaml` since his is already pushed to the main branch.